src/vs/workbench/services/themes/common/colorThemeData.ts

```diff
@@ -587,6 +587,22 @@ export class ColorThemeData implements IWorkbenchColorTheme {
 	// constructors
 
 	static createUnloadedThemeForThemeType(themeType: ColorScheme, colorMap?: { [id: string]: string }): ColorThemeData {
+		if (!colorMap) {
+			if (themeType === ColorScheme.LIGHT) {
+				colorMap = {
+					'activityBar.background': '#f8f8f8',
```

Comment:
I'm wondering where these colours come from, and when they are displayed? (I'm not sure what it means to have a falsy value for `colorMap`)

Reply:
They come from the new default themes: https://github.com/microsoft/vscode/blob/d53b4bdb0be6b83f5fe997399a92140adee38ec7/extensions/theme-defaults/themes/dark_plus_experimental.json
and
https://github.com/microsoft/vscode/blob/d53b4bdb0be6b83f5fe997399a92140adee38ec7/extensions/theme-defaults/themes/light_plus_experimental.json

I hard

The problem is that when VS Code starts up the very first time, we don't have any persisted colors, so we show the default values of all colors. These do not match the new default themes. 
It takes a short time to load the actual theme from the fiels above. So for a short time you see the wrong colors. To avoid flickering I hardcode the new colors that are visible right away: the status bar and the activity bar.

Note that we can't change the default values of color as this would break other themes that rely on these defaults.